feat(trace-view): Include errors in the full trace

wmak opened the pull request:
- This adds a list of errors to the full trace view
- This also cleans up the handle_events code a bit and moves
  handle_issues into its own function so that it can be easily
  reused by the trace view

wmak committed:
feat(trace-view): Include errors in the full trace

- This adds a list of errors to the full trace view
- This also cleans up the handle_events code a bit and moves
  handle_iss<Plug>_ues into its own function so that it can be easily
  reused by the trace view

wmak committed:
Merge branch 'master' into feat/trace-view-errors

wmak committed:
ref: Only return errors for users with summary enabled

wmak committed:
ref: Cleaning up get_error_map

wmak commented:
TODO: Realized I forgot to add event.type:default to this.

Zylphrex reviewed (approved):
LGTM.

We do need to look into `event.type:default` and `event.type:csp` to see if they are something we should be including in the results here.

wmak committed:
fix: Supporting more error types (default/csp)

- Also adding span to the error response as well so we can associate
  using the trace endpoint instead of an extra eventsv2 query

wmak commented:
Based on my exploration from: https://github.com/getsentry/sentry/pull/24311 ended up going with `!event.type:transaction` also realized that I had to include the span in the error response so that we could eventually remove the reliance on a call to eventsv2.

wmak committed:
Merge branch 'master' into feat/trace-view-errors

wmak committed:
Merge branch 'master' into feat/trace-view-errors

wmak committed:
fix: Probably should've ran these tests, woops
